eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Gateway/ExceptionConversion.php

```diff
@@ -36,6 +36,17 @@ public function __construct(Gateway $innerGateway)
         $this->innerGateway = $innerGateway;
     }
 
+    public function setTable($name)
+    {
+        try {
+            return $this->innerGateway->setTable($name);
+        } catch (DBALException $e) {
+            throw new \RuntimeException('Database error', 0, $e);
+        } catch (PDOException $e) {
+            throw new \RuntimeException('Database error', 0, $e);
```

Comment:
True, but we still do this in layers like this one, as a matter of not caring how the underlying implementation behaves. For example we do the similar with return values in `SignalSlot` Repository services and in SPI Cache storage handlers.